File: php/program/BarangEdisiTerbatas.php
<?php
require_once 'BarangPremium.php';

// Class Level 3: BarangEdisiTerbatas
class BarangEdisiTerbatas extends BarangPremium {
    private $nomorSeri;
    private $jumlahProduksi;

    public function __construct($id, $nama, $merek, $harga, $fotoProduk, $material, $fiturEksklusif, $nomorSeri, $jumlahProduksi) {
        parent::__construct($id, $nama, $merek, $harga, $fotoProduk, $material, $fiturEksklusif);
        $this->nomorSeri = $nomorSeri;
        $this->jumlahProduksi = $jumlahProduksi;
    }

    // Getters
    public function getNomorSeri() { return $this->nomorSeri; }
    public function getJumlahProduksi() { return $this->jumlahProduksi; }
    
    // Setters
    public function setNomorSeri($seri) { $this->nomorSeri = $seri; }
    public function setJumlahProduksi($jumlah) { $this->jumlahProduksi = $jumlah; }

    public function __destruct() {
        parent::__destruct();
    }
}
?>

File: cpp/program/BarangEdisiTerbatas.cpp

// Class Level 3: BarangEdisiTerbatas
class BarangEdisiTerbatas : public BarangPremium {
private:
    string nomorSeri;
    int jumlahProduksi;

public:
    BarangEdisiTerbatas(int id, string nama, string merek, double harga, string material, string fiturEksklusif, string nomorSeri, int jumlahProduksi)
        : BarangPremium(id, nama, merek, harga, material, fiturEksklusif), nomorSeri(nomorSeri), jumlahProduksi(jumlahProduksi) {}
    
    ~BarangEdisiTerbatas() override {
    }

    string getNomorSeri() const { return nomorSeri; }
    int getJumlahProduksi() const { return jumlahProduksi; }

    void setNomorSeri(const string& seri) { this->nomorSeri = seri; }
    void setJumlahProduksi(int jumlah) { this->jumlahProduksi = jumlah; }

    void display(const map<string, int>& colWidths) const override {
        BarangPremium::display(colWidths);
        cout << setw(colWidths.at("Nomor Seri")) << left << getNomorSeri() << " | "
             << setw(colWidths.at("Jumlah Produksi")) << left << getJumlahProduksi() << " | " << endl;
    }
};
